tools/matc/src/matc/CommandlineConfig.cpp

```diff
@@ -115,8 +119,35 @@ CommandlineConfig::CommandlineConfig(int argc, char** argv) : Config(), mArgc(ar
     mIsValid = parse();
 }
 
+static void parseDefine(std::string defineString,
+        std::unordered_map<std::string, std::string>& defines) {
+    const char* const defineArg = defineString.c_str();
+    const size_t length = defineString.length();
+
+    const char* p = defineArg;
+    const char* end = p + length;
+
+    while (p < end && *p != '=') {
+        p++;
+    }
+
+    if (*p == '=') {
+        if (p == defineArg || p + 1 >= end) {
+            // Edge-cases, missing define name or value.
+            return;
+        }
+        std::string def(defineArg, p - defineArg);
+        defines.emplace(def, p + 1);
+        return;
+    }
+
+    // No explicit assignment, use a default value of 1.
+    std::string def(defineArg, p - defineArg);
+    defines.emplace(def, "1");
```

Comment:
We could just use empty string as well. That's enough for #ifdef or #if defined(…) to work.

Reply:
True, this follows Clang's convention of setting it explicitly to 1.